src/parser/shared/metrics/activeTime.ts

```diff
@@ -0,0 +1,70 @@
+import { AnyEvent, EventType } from 'parser/core/Events';
+import metric, { Info } from 'parser/core/metric';
+import { ClosedTimePeriod, duration, intersection, union } from 'parser/core/timePeriods';
+
+// TODO possible extensions:
+//   * channel and GCD time periods marked with spell, allowing additional filtering
+
+/** Time periods the player was channeling during the encounter */
+export const channelTimePeriods = metric((events: AnyEvent[], info: Info): ClosedTimePeriod[] => {
```

Comment:
I think it would be nice to put this in a separate file. It can probably be used for multiple purposes.

Reply:
Ah yeah I wasn't sure how much you wanted to break this up by file. I had been putting logically connected functions together even though here you can see each outward facing function is exported. Is there any mechanical difference to separating them or is it just style?

Reply:
Just style. I like to think of separate files as separate features that can be used by itself, and when channelTimePeriods can have its own live outside of active time, I think it makes sense as a separate file.